app/src/layouts/kanban/index.ts

```diff
@@ -0,0 +1,596 @@
+					[gField]: group.id,
+				});
+			}
+		}
+
+		function parseUrl(file: Record<string, any>) {
+			if (!file || !file.type) return;
+			if (file.type.startsWith('image') === false) return;
+			if (file.type.includes('svg')) return;
+
+			const fit = crop.value ? '&width=250&height=150' : `&key=system-medium-contain`;
+
+			const url = getRootPath() + `assets/${file.id}?modified=${file.modified_on}` + fit;
+			return addTokenToURL(url);
+		}
+
+		function useLayoutOptions() {
+			const groupField = createViewOption<string | null>('groupField', fieldGroups.value.group[0]?.field ?? null);
+			const groupTitle = createViewOption<string | null>('groupTitle', null);
+			const dateField = createViewOption<string | null>('dateField', fieldGroups.value.date[0]?.field ?? null);
+			const tagsField = createViewOption<string | null>('tagsField', fieldGroups.value.tags[0]?.field ?? null);
+			const userField = createViewOption<string | null>('userField', fieldGroups.value.user[0]?.field ?? null);
+			const titleField = createViewOption<string | null>('titleField', fieldGroups.value.title[0]?.field ?? null);
+			const textField = createViewOption<string | null>('textField', fieldGroups.value.text[0]?.field ?? null);
+			const showUngrouped = createViewOption<boolean>('showUngrouped', false);
+			const imageSource = createViewOption<string | null>('imageSource', fieldGroups.value.file[0]?.field ?? null);
+			const crop = createViewOption<boolean>('crop', true);
+
+			const selectedGroup = computed(() => fieldGroups.value.group.find((group) => group.field === groupField.value));
+
+			watch(groupField, () => {
+				groupTitle.value = null;
+			});
+
+			const userFieldJunction = computed(() => {
+				if (userField.value === null) return;
+
+				return relationsStore.relations.find(
+					(relation) =>
+						relation.meta?.one_collection === props.collection &&
+						relation.meta.one_field === userField.value &&
+						relation.meta.junction_field !== null
+				);
+			});
+
+			const userFieldType = computed(() => {
+				if (userField.value === null) return;
+				return userFieldJunction.value !== undefined ? 'm2m' : 'm2o';
+			});
+
+			return {
+				groupField,
+				groupTitle,
+				imageSource,
+				selectedGroup,
+				titleField,
+				textField,
+				crop,
+				dateField,
+				tagsField,
+				userField,
+				showUngrouped,
+				userFieldJunction,
+				userFieldType,
+			};
+
+			function createViewOption<T>(key: keyof LayoutOptions, defaultValue: any) {
+				return computed<T>({
+					get() {
+						return layoutOptions.value?.[key] !== undefined ? layoutOptions.value?.[key] : defaultValue;
+					},
+					set(newValue: T) {
+						layoutOptions.value = {
+							...layoutOptions.value,
+							[key]: newValue,
+						};
+					},
+				});
+			}
+		}
+
+		function useGrouping() {
+			const isRelational = computed(() => {
+				return !!selectedGroup.value?.meta?.options?.choices === false;
```

Comment:
```suggestion
				return selectedGroup.value?.meta?.options?.choices === false;
```

Reply:
I believe the original double negation was meant to handle `!!undefined === false` which will be true, but now `undefined === false` will be false.

But I've opted to do `!selectedGroup.value?.meta?.options.choices` instead so that `!undefined` would indeed be true, and it'll be false if `choices` is an array, which seems logical. Do let me know if that's still incorrect 👍